### PR TITLE
feat: commit pending proposals FS-703

### PR DESCRIPTION
**Pull request by typfel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-703" title="FS-703" target="_blank">TaskFS-703</a>  [iOS] Automatically Commit PendingProposals
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Commit incoming pending proposals as described here: https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/601522340/Use+Case+Committing+Pending+Valid+Proposals+MLS

### Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/566
- [x] https://github.com/wireapp/wire-ios-data-model/pull/1405

#### Test Coverage

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/UserSession/ZMUserSession.swift [hunk 1]
```diff
@@ -566,6 +566,15 @@ extension ZMUserSession: ZMSyncStateDelegate {
...
         }
 
+        let mlsController = syncContext.mlsController
+        Task {
+            do {
+                try await mlsController?.commitPendingProposals()
+            } catch {
+                Logging.mls.error("Failed to commit pending proposals: \(String(describing: error))")
+            }
+        }
```
Comment (David-Henner) [inline Source/UserSession/ZMUserSession.swift:576]: I'd suggest to put this in its own method

File: Tests/Source/Mocks/MockMLSController.swift [hunk 2]
```diff
@@ -54,7 +54,7 @@ class MockMLSController: MLSControllerProtocol {
         fatalError("not implemented")
     }
 
-    func addGroupPendingJoin(_ group: MLSGroupID) {
+    func addGroupPendingJoin(_ group: MLSGroup) {
```
Comment (David-Henner) [inline Tests/Source/Mocks/MockMLSController.swift:57]:
```suggestion
    func addGroupPendingJoin(_ group: MLSGroupID) {
```

File: Tests/Source/Mocks/MockMLSController.swift [hunk 3]
```diff
@@ -63,4 +63,16 @@ class MockMLSController: MLSControllerProtocol {
...
+    }
+
+    func scheduleCommitPendingProposals(groupID: MLSGroupID, at commitDate: Date) {
+        fatalError("not implemented")
+    }
+
+    func addGroupPendingJoin(_ group: MLSGroupID) {
+        fatalError("not implemented")
+    }
+
```
Comment (David-Henner) [inline Tests/Source/Mocks/MockMLSController.swift:77]:
```suggestion
```

File: Cartfile.resolved [hunk 4]
```diff
@@ -15,7 +15,7 @@ github "wireapp/wire-ios-protos" "32.0.0"
 github "wireapp/wire-ios-system" "38.0.0"
 github "wireapp/wire-ios-testing" "29.0.0"
 github "wireapp/wire-ios-utilities" "47.0.0"
-github "wireapp/wire-ios-data-model" "feat/mls"
+github "wireapp/wire-ios-data-model" "feat/commit-pending-proposals"
```
Comment (johnxnguyen) [inline Cartfile.resolved:18]:
```suggestion
github "wireapp/wire-ios-data-model" "feat/mls"
```

File: Cartfile.resolved [hunk 5]
```diff
@@ -15,7 +15,7 @@ github "wireapp/wire-ios-protos" "32.0.0"
...
 github "wireapp/wire-ios-mocktransport" "feat/mls"
-github "wireapp/wire-ios-request-strategy" "feat/mls"
+github "wireapp/wire-ios-request-strategy" "feat/commit-pending-proposals"
```
Comment (johnxnguyen) [inline Cartfile.resolved:20]:
```suggestion
github "wireapp/wire-ios-request-strategy" "feat/mls"
```

File: Source/UserSession/ZMUserSession.swift [hunk 6]
```diff
@@ -588,6 +589,17 @@ extension ZMUserSession: ZMSyncStateDelegate {
         }
     }
 
+    private func commitPendingProposalsIfNeede() {
```
Comment (David-Henner) [inline Source/UserSession/ZMUserSession.swift:592]:
```suggestion
    private func commitPendingProposalsIfNeeded() {
```

File: Source/UserSession/ZMUserSession.swift [hunk 7]
```diff
@@ -566,6 +566,7 @@ extension ZMUserSession: ZMSyncStateDelegate {
             self?.notifyThirdPartyServices()
         }
 
+        commitPendingProposalsIfNeede()
```
Comment (David-Henner) [inline Source/UserSession/ZMUserSession.swift:569]:
```suggestion
        commitPendingProposalsIfNeeded()
```

**Review by David-Henner — APPROVED** [review]

**Pull request merged**